bcgov/mds
# MDS-4045: Update Admin Amendment flow to use new "is_generated_in_core" flag

# Main

The current flow for permit amendments changes to follow the next:

1. Source Amendment was generated in Core - populate new draft permit with previous amendment conditions. (The resulting permit amendment should have ‘is_generated_in_core = True’)

2. Source Amendment was not generated in Core - allow users to bypass permit writing and upload the permit pdf.  (The resulting permit amendment should have ‘is_generated_in_core = False’)
 
3. Source Amendment was not generated in Core - allow users to draft a permit_amendment and populate with standard permit conditions. (The resulting permit amendment should have ‘is_generated_in_core = True)

# Other
If application tab is not having assign the inspectors, disable the following tabs.

# How to test
Preparation
1.- Create a permit type placer with issue date on December 1st:
![image](https://user-images.githubusercontent.com/10526131/146976138-9b8784b6-f204-40ff-9771-fc61e05eb772.png)

2.- Go to Permits & Approvals > Applications. Then, under Administrative Amendments  create a new Administrative Amendment referring to the previous permit created which Source Amendment was issued by December 1st :
![image](https://user-images.githubusercontent.com/10526131/146976196-e5b0ab00-a137-42b5-b538-29ceb3d6670c.png)

3.- Select draft tab and click on “Start Draft Permit” button. Because this permit was not generated by Core (not issued by Core), then we should find the following modal:
![image](https://user-images.githubusercontent.com/10526131/146976420-7c8339d9-0e43-467d-9b48-a9f84464d542.png)

Then click on “Proceed” button.

The permits not generated in Core have two flows, Write Permit in Core (which creates a permit generated in Core) and Upload Permit (which creates a permit NOT generated in Core). 

4.- Select Upload Permit flow and click on “Start Draft Permit” button:
![image](https://user-images.githubusercontent.com/10526131/146976526-e12ee204-c16f-4860-a732-40018054a7cf.png)

5.- Upload a PDF file and address all issues under Process tab:
![image](https://user-images.githubusercontent.com/10526131/146976648-81ea8b88-7809-46d1-9d85-b28c80eec822.png)

6.- Go to Process dropdown and select “Issue Permit”. Generate letter and Issue permit with **issue date on December 2nd**:
![image](https://user-images.githubusercontent.com/10526131/146976725-591f278d-97ff-4bac-a0ec-882e2777cdf7.png)
![image](https://user-images.githubusercontent.com/10526131/146977128-a977b91a-7c38-4a48-b772-25ebe401de24.png)

7.- Create a new Administrative Amendment considering as a base the previous created permit P-202111221 and Issue Date December 1st (as the previous created amendment):
![image](https://user-images.githubusercontent.com/10526131/146977172-9dd77359-a57d-4326-9fe8-def9c5da99a4.png)

8.- Select draft tab and click on “Start Draft Permit” button. Because this permit was not generated by Core (not issued by Core), then we should find the same modal as the previous amendment:
![image](https://user-images.githubusercontent.com/10526131/146977503-16c3f4db-d263-4585-849c-76bab4943df4.png)

Then click on “Proceed” button.

9.- Select Write Permit in Core which will create a permit generated in Core. Click on “Start Draft Permit” button:
![image](https://user-images.githubusercontent.com/10526131/146977670-dc3af52a-0c21-4976-998c-c5278ae6877e.png)

10.- Address all issues under Process to be able to issue a permit:
![image](https://user-images.githubusercontent.com/10526131/146977716-e9309f2f-98a7-4c0f-acaf-e3744d4136d1.png)

11.- Go to Process dropdown and select “Issue Permit”. Then Generate the letter and issue a permit with issue date on December 3rd:
![image](https://user-images.githubusercontent.com/10526131/146977759-7c1ec571-9fb5-4739-90cc-ee8b9d7273e3.png)

To recap, we have two source amendments:
- Dec 02, 2021 -> Not generated in Core (Upload Permit flow)
- Dec 03, 2021 -> Generated in Core (Write permit in Core)

The first amendment permit should show the following modal during “Start Draft Permit”:
![image](https://user-images.githubusercontent.com/10526131/146977849-c27ed8da-c4a3-4455-84da-ed28e07e6312.png)

Then, finish the usual steps and issue the permit.

The second amendment should show the following modal during “Start Draft Permit”:
![image](https://user-images.githubusercontent.com/10526131/146977903-dd963903-5a2d-43ce-be52-32bc023f1ac8.png)

To confirm that an amendment is inheriting conditions from its parent permit follow the next steps: 
1.- Create an additional amendment with December 03rd as Source Amendment by issue date and modify the original conditions. Then, click on Start Draft Permit button and select Conditions. Under Conditions, modify the first condition:
![image](https://user-images.githubusercontent.com/10526131/146977994-3093cf85-db9e-4880-a1aa-0147d0839392.png)

2.- Fix all issues and Issue the permit with Issue Date on December 15th:
![image](https://user-images.githubusercontent.com/10526131/146978040-6871197c-b48b-46f3-ba6a-0dba43f8d5b4.png)

3.- Create another amendment by clicking on Add Administrative Amendment and select Source Amendment by issue date on December 15th (previous amendment with the first condition modified):
![image](https://user-images.githubusercontent.com/10526131/146978073-7c2a96cd-181d-43e1-b6df-49c646428540.png)

4.- Under Draft tab, click on Start Draft Permit button and go to Conditions to verify we are getting the modified condition instead of the original:
![image](https://user-images.githubusercontent.com/10526131/146978120-8256199d-59fe-4fdf-8a90-9d9afd0baa07.png)

Additionally, if inspectors are not assigned, the user cannot go to the other tabs:
![image](https://user-images.githubusercontent.com/10526131/146978171-a9f0269d-dc28-4d52-b9d6-86d8d6dc4734.png)

# Notes
https://bcmines.atlassian.net/browse/MDS-4045
